src/GafferScene/CopyPrimitiveVariables.cpp

```diff
@@ -0,0 +1,208 @@
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "GafferScene/CopyPrimitiveVariables.h"
+
+#include "GafferScene/SceneAlgo.h"
+
+#include "Gaffer/ArrayPlug.h"
+
+#include "IECoreScene/Primitive.h"
+
+using namespace std;
+using namespace IECore;
+using namespace IECoreScene;
+using namespace Gaffer;
+using namespace GafferScene;
+
+GAFFER_GRAPHCOMPONENT_DEFINE_TYPE( CopyPrimitiveVariables );
+
+size_t CopyPrimitiveVariables::g_firstPlugIndex = 0;
+
+CopyPrimitiveVariables::CopyPrimitiveVariables( const std::string &name )
+	:	Deformer( name, 2, 2 )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+	addChild( new StringPlug( "primitiveVariables", Plug::In, "" ) );
+	addChild( new StringPlug( "copyFrom" ) );
+}
+
+CopyPrimitiveVariables::~CopyPrimitiveVariables()
+{
+}
+
+Gaffer::StringPlug *CopyPrimitiveVariables::primitiveVariablesPlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex );
+}
+
+const Gaffer::StringPlug *CopyPrimitiveVariables::primitiveVariablesPlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex );
+}
+
+Gaffer::StringPlug *CopyPrimitiveVariables::copyFromPlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 1 );
+}
+
+const Gaffer::StringPlug *CopyPrimitiveVariables::copyFromPlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 1 );
+}
+
+bool CopyPrimitiveVariables::affectsProcessedObject( const Gaffer::Plug *input ) const
+{
+	return Deformer::affectsProcessedObject( input ) ||
+		input == inPlugs()->getChild<ScenePlug>( 1 )->objectPlug() ||
```

Comment:
I was following the example of CopyAttributes (and the spec from the ticket) here, but I agree. I'm loathe to have this and CopyAttributes differ though - separate followup PR to change them both together? Either way, how about just `source` for the plug name?

On a related note, @themissingcow mentioned that his first impression of `copyFrom` was that it would scope operations to a subtree somehow, a bit like `Isolate.from`. Would `sourceLocation` be better?

Reply:
`source` and `sourceLocation` sound good to me. I also didn't like `copyFrom` but didn't have a better suggestion.

> separate followup PR to change them both together?

sure

Reply:
I went ahead and did the renaming in this PR in the end. Since there weren't any unresolved comments I've rebased the CopyPrimitiveVariables changes into the original commits ready for merging. And I've adjusted both CopyOptions and CopyAttributes to match in 8f434aa.